#!/usr/bin/env python
PACKAGE = "gazebo_ros_motors"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("velocity_noise",               double_t, 1, "Encoder measurement standard deviation", 0.0, 0.0, 10.0)
gen.add("motor_nominal_voltage",        double_t, 1, "Motor nominal voltage", 12.0, 0.0, 120.0)
gen.add("electric_resistance",          double_t, 1, "Motor electric resistance", 1.0, 0.0, 20.0)
gen.add("electric_inductance",          double_t, 1, "Motor electric inductance", 0.001, 0.0, 1.0)
gen.add("moment_of_inertia",            double_t, 1, "Motor moment of inertia", 0.001, 0.0, 1.0)
gen.add("armature_damping_ratio",       double_t, 1, "Motor armature damping ratio", 0.0001, 0.0, 1.0)
gen.add("electromotive_force_constant", double_t, 1, "Motor electromotive force constant", 0.01, 0.0, 1.0)

exit(gen.generate(PACKAGE, "gazebo_ros_motors", "motorModel"))
